public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -203,23 +206,57 @@ export function createDashboardSceneFromDashboardModel(oldModel: DashboardModel)
     });
   }
 
-  let controls: SceneObject[] = [
-    new VariableValueSelectors({}),
-    ...filtersSets,
-    new SceneDataLayerControls(),
-    new SceneControlsSpacer(),
-  ];
+  const variableControls = new SceneFlexItem({
+    body: new SceneFlexLayout({
+      direction: 'row',
+      md: {
+        direction: 'row',
+      },
+      wrap: 'wrap',
+      children: [
+        new VariableValueSelectors({}),
+        ...filtersSets,
+        new SceneDataLayerControls(),
+        new SceneControlsSpacer(),
+        new DashboardLinksControls({
+          links: oldModel.links,
+          dashboardUID: oldModel.uid,
+        }),
+      ],
+    }),
+  });
```

Comment:
not 100% sure about this, will make it a lot more complex to update the controls (say when links change, or something else change that require us to update controls). 

is there any way we can handle this layout / css issue in the DashboardSceneRenderer (create some kind of component that renders controls that can adapt the css and markup depending on what controls are present)

Reply:
Wanted to quickly experiment with a layout that could work. I think DashboardScene can have a custom controls wrapper. Will do that.